inc/Engine/Preload/Fonts.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace WP_Rocket\Engine\Preload;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Fonts preload.
+ *
+ * @since 3.6
+ */
+class Fonts implements Subscriber_Interface {
+
+	/**
+	 * WP Rocket Options instance.
+	 *
+	 * @since 3.6
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @param Options_Data $options WP Rocket Options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'wp_head' => [ 'preload_fonts' ],
+		];
+	}
+
+	/**
+	 * Add the required <link/> tags used to preload fonts.
+	 *
+	 * @since 3.6
+	 */
+	public function preload_fonts() {
+		$fonts = $this->options->get( 'preload_fonts', [] );
+		$fonts = array_map( [ $this, 'sanitize_font' ], $fonts );
+		$fonts = array_filter( $fonts );
+
+		if ( empty( $fonts ) ) {
+			return;
+		}
+
+		$home = untrailingslashit( get_option( 'home' ) );
```

Comment:
```suggestion
		$site_url = get_rocket_parse_url( site_url() );
		$home   = $site_url['scheme'] . '://' . $site_url['host']:
```

How about this instead?

I understand the idea of using `get_option( 'home' )` directly instead of `home_url()` because of the filter, but I also feel `site_url()` is a good way to retrieve what we want. This is also what is used in other parts of the plugin.

Reply:
`get_option( 'home' )`, `site_url()`, `home_url()`, all three are used in the plugin and `home_url()` is the most used, but this must not be one of the criteria for deciding.

So here are my thoughts, you may have right about using `site_url()`:
1- Among other places, fonts may be located in themes or plugins.
2- `WP_PLUGIN_URL` from WP’s core:
```php
if ( ! defined( 'WP_CONTENT_URL' ) ) {
	define( 'WP_CONTENT_URL', get_option( 'siteurl' ) . '/wp-content' );
}

// ...

if ( ! defined( 'WP_PLUGIN_URL' ) ) {
	define( 'WP_PLUGIN_URL', WP_CONTENT_URL . '/plugins' );
}
```

So, `WP_PLUGIN_URL` uses `WP_CONTENT_URL`, which uses `get_option( 'siteurl' )`.
3- `get_site_url()` also uses `get_option( 'siteurl' )` obviously.

So I think `site_url()` is the safest one to use.

Reply:
Done in https://github.com/wp-media/wp-rocket/pull/2563/commits/242860e25cd7f02c2300a17bef508d3165438c24.